chain/filecoin/account.go

```diff
@@ -0,0 +1,294 @@
+		return fmt.Errorf("expected signature to be 65 bytes, got %v bytes", len(signatures[0]))
+	}
+
+	// we are sure there are exactly 65 bytes in the signature, so this is safe
+	sig := [65]byte{}
+	for i := range sig {
+		sig[i] = signatures[0][i]
+	}
+
+	tx.signature = pack.NewBytes65(sig)
+	return nil
+}
+
+// Serialize the transaction into bytes. Generally, this is the format in
+// which the transaction will be submitted by the client.
+func (tx Tx) Serialize() (pack.Bytes, error) {
+	buf := new(bytes.Buffer)
+	if err := tx.msg.MarshalCBOR(buf); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// TxBuilder represents a transaction builder that builds transactions to be
+// broadcasted to the filecoin network. The TxBuilder is configured using a
+// gas price and gas limit.
+type TxBuilder struct {
+	gasFeeCap  pack.U256
+	gasLimit   pack.U256
+	gasPremium pack.U256
+}
+
+// NewTxBuilder creates a new transaction builder.
+func NewTxBuilder(gasFeeCap, gasLimit, gasPremium pack.U256) TxBuilder {
+	return TxBuilder{gasFeeCap: gasFeeCap, gasLimit: gasLimit, gasPremium: gasPremium}
+}
+
+// BuildTx receives transaction fields and constructs a new transaction.
+func (txBuilder TxBuilder) BuildTx(from, to address.Address, value, nonce, _, _ pack.U256, payload pack.Bytes) (account.Tx, error) {
+	filfrom, err := filaddress.NewFromString(string(from))
+	if err != nil {
+		return nil, fmt.Errorf("bad from address '%v': %v", from, err)
+	}
+	filto, err := filaddress.NewFromString(string(to))
+	if err != nil {
+		return nil, fmt.Errorf("bad to address '%v': %v", to, err)
+	}
+	methodNum := abi.MethodNum(0)
+	if len(payload) > 0 {
+		methodNum = abi.MethodNum(payload[0])
+		payload = payload[1:]
+	}
+	return &Tx{
+		msg: types.Message{
+			Version:    types.MessageVersion,
+			From:       filfrom,
+			To:         filto,
+			Value:      big.Int{Int: value.Int()},
+			Nonce:      nonce.Int().Uint64(),
+			GasFeeCap:  big.Int{Int: txBuilder.gasFeeCap.Int()},
+			GasLimit:   txBuilder.gasLimit.Int().Int64(),
+			GasPremium: big.Int{Int: txBuilder.gasPremium.Int()},
+			Method:     methodNum,
+			Params:     payload,
+		},
+		signature: pack.Bytes65{},
+	}, nil
+}
+
+// ClientOptions are used to parameterise the behaviour of the Client.
+type ClientOptions struct {
+	MultiAddress string
+	AuthToken    string
+}
+
+// DefaultClientOptions returns ClientOptions with the default settings. These
+// settings are valid for use with the default local deployment of the
+// multichain. In production, the multi-address and authentication token should
+// be changed.
+func DefaultClientOptions() ClientOptions {
+	return ClientOptions{
+		MultiAddress: DefaultClientMultiAddress,
+		AuthToken:    DefaultClientAuthToken,
+	}
+}
+
+// WithAddress returns a modified version of the options with the given API
+// multi-address.
+func (opts ClientOptions) WithAddress(multiAddr string) ClientOptions {
+	opts.MultiAddress = multiAddr
```

Comment:
As per one of the previous comments, I am not sure this is actually a multi-address. Pretty sure it is just an RPC URL.